source/extensions/common/proxy_protocol/proxy_protocol_header.cc

```diff
@@ -118,6 +127,27 @@ void generateV2LocalHeader(Buffer::Instance& out) {
   out.add(addr_fam_protocol_and_length, 4);
 }
 
+void generateV2Header(const Network::ProxyProtocolData& prox_proto_data, Buffer::Instance& out) {
+  uint16_t extension_length = 0;
+  for (auto& tlv : prox_proto_data.tlv_vector_) {
```

Comment:
not auto&& ?

Reply:
Ah. Good catch!

Reply:
Why not the following for() loop as well?